pkg/networkservice/common/discover/server.go

```diff
@@ -106,24 +106,13 @@ func (d *discoverCandidatesServer) Request(ctx context.Context, request *network
 }
 
 func (d *discoverCandidatesServer) Close(ctx context.Context, conn *networkservice.Connection) (*empty.Empty, error) {
-	if clienturlctx.ClientURL(ctx) != nil {
-		return next.Server(ctx).Close(ctx, conn)
-	}
-
-	nseName := conn.GetNetworkServiceEndpointName()
-	if nseName == "" {
+	if conn.GetNetworkServiceEndpointName() == "" {
 		// If it's an existing connection, the NSE name should be set. Otherwise, it's probably an API misuse.
 		return nil, errors.Errorf("network_service_endpoint_name is not set")
 	}
-	nse, err := d.discoverNetworkServiceEndpoint(ctx, nseName)
-	if err != nil {
-		return nil, err
-	}
-	u, err := url.Parse(nse.Url)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-	return next.Server(ctx).Close(clienturlctx.WithClientURL(ctx, u), conn)
+
+	// Connect server already has a client URL for the conn.Id, so we don't need to set it here.
```

Comment:
I don't think that discover server should know about connect server.

Reply:
Removed this commit.